src/PointwiseFunctions/Hydro/InitialData/IrrotationalBns.cpp

```diff
@@ -0,0 +1,270 @@
+  ::tenex::evaluate<ti::k>(
+      result, rotational_shift(ti::J) *
+                      derivative_rotational_shift_over_lapse(ti::j, ti::I) *
+                      spatial_metric(ti::i, ti::k) / lapse() +
+                  rotational_shift(ti::I) * spatial_metric(ti::i, ti::k) *
+                      derivative_rotational_shift_over_lapse(ti::j, ti::J) /
+                      lapse());
+}
+
+template <typename DataType>
+tnsr::i<DataType, 3> divergence_rotational_shift_stress(
+    const tnsr::I<DataType, 3>& rotational_shift,
+    const tnsr::iJ<DataType, 3>& derivative_rotational_shift_over_lapse,
+    const Scalar<DataType>& lapse,
+    const tnsr::ii<DataType, 3>& spatial_metric) {
+  tnsr::i<DataType, 3> buffer{};
+  divergence_rotational_shift_stress(make_not_null(&buffer), rotational_shift,
+                                     derivative_rotational_shift_over_lapse,
+                                     lapse, spatial_metric);
+  return buffer;
+}
+
+template <typename DataType>
+void enthalpy_density_squared(
+    const gsl::not_null<Scalar<DataType>*> result,
+    const tnsr::I<DataType, 3>& rotational_shift, const Scalar<DataType>& lapse,
+    const tnsr::i<DataType, 3>& velocity_potential_gradient,
+    const tnsr::II<DataType, 3>& inverse_spatial_metric,
+    const double euler_enthalpy_constant) {
+  return tenex::evaluate<>(
+      result,
+      square((euler_enthalpy_constant +
+              rotational_shift(ti::I) * velocity_potential_gradient(ti::i)) /
+             lapse()) -
+          velocity_potential_gradient(ti::i) *
+              velocity_potential_gradient(ti::j) *
+              inverse_spatial_metric(ti::I, ti::J));
+}
+
+template <typename DataType>
+Scalar<DataType> enthalpy_density_squared(
+    const tnsr::I<DataType, 3>& rotational_shift, const Scalar<DataType>& lapse,
+    const tnsr::i<DataType, 3>& velocity_potential_gradient,
+    const tnsr::II<DataType, 3>& inverse_spatial_metric,
+    const double euler_enthalpy_constant) {
+  Scalar<DataType> buffer{};
+  enthalpy_density_squared(make_not_null(&buffer), rotational_shift, lapse,
+                           velocity_potential_gradient, inverse_spatial_metric,
+                           euler_enthalpy_constant);
+  return buffer;
+}
+
+template <typename DataType>
+void spatial_rotational_killing_vector(
+    const gsl::not_null<tnsr::I<DataType, 3>*> result,
+    const tnsr::I<DataType, 3>& x,
+    const Scalar<DataType>& local_angular_velocity_around_z,
+    const Scalar<DataType>& sqrt_det_spatial_metric) {
+  // Cross product involves volume element in arbitrary coordinates
+  result->get(0) = -get(sqrt_det_spatial_metric) * get<1>(x) *
+                   get(local_angular_velocity_around_z);
+  result->get(1) = get(sqrt_det_spatial_metric) * get<0>(x) *
+                   get(local_angular_velocity_around_z);
+  result->get(2) = make_with_value<DataType>(sqrt_det_spatial_metric, 0.0);
```

Comment:
For this function, use `set_number_of_grid_points(result, sqrt_det_spatial_metric)` (from `Utilities/SetNumberOfGridPoints.hpp`) at the start of the function and just assign a plain `0.0` to this component.  Using `make_with_value` does an extra memory allocation and could break things if `result` was part of a Variables (and was therefore non-owning).  For most of the other functions the tenex code will handle the resizing.